### ISSUE TITLE
[1.12.2] Ghast Spawning [Custom Spawner Issue]

### ISSUE THREAD

**Issue by EvilDylan** [DrZhark/mocreatures#81]
With only this mod and Custom Spawner installed:
Custom Spawner appears to have a tendency to spawn mobs partially or fully in walls. I have seen this happen with spiders, alpacas, and most importantly, ghasts.
When a ghast is suffocating, it fills the crisp nether air with ear-splitting shrieks of eternal suffering until it eventually dies. This is incredibly annoying and disorienting.

**Comment by darchitectlp** [comment]
I'm not sure it's CustomMobSpawner... I didn't install it, and I've seen dolphins and fish spawn in sand and a few other bugged spawns.

**Comment by durand1w** [comment]
Can confirm this is still an issue on 1.12 version of Mo'Creatures and CMS.  This has been an issue on previous releases as well.  Is it possible that the spawning checks are not verifying that there are Blocks.AIR at the position and also above the position for regular mobs?  For Ghasts it probably requires a much larger position check for Air blocks.

We also have modded blocks that mobs spawn in regularly.  I think this is related to the same area of code where the modded block darkens the vanilla block below it.  CMS only checks the lightlevel and if the block is not a full block (stairs, for example)?

**Comment by darchitectlp** [comment]
I think the spawning rules that are causing this are built into Mo Creatures, itself. As I said, I play without CMS installed, and a lot of Mo Creatures mobs spawn in blocks.

**Comment by durand1w** [comment]
Yeah, I can confirm the issue with Mobs spawning inside Redstone Blocks still exists as well.  (screenshot attached).

![2018-07-21_08 46 20](https://user-images.githubusercontent.com/4107063/43051995-92ac05bc-8de6-11e8-826a-e46b7521eb43.png)
